compiler/AST/primitive.cpp

```diff
@@ -151,49 +153,49 @@ returnInfoCast(CallExpr* call) {
   if (t2->symbol->hasFlag(FLAG_WIDE_REF))
     if (wideRefMap.get(t1))
       t1 = wideRefMap.get(t1);
-  return t1;
+  return QualifiedType(t1); // what should qual be here?
```

Comment:
In Chapel, an explicit conversion - and perhaps an implicit conversion as well - produces an r-value. So I'd say it should be qVal.